pkg/lib/cockpit-po-plugin.js

```diff
@@ -105,24 +78,36 @@ module.exports = class {
                 ? this.filterMessages(po_file, compilation)
                 : fs.readFileSync(po_file);
 
-            const jsonData = po2json.parse(poData);
-            const plurals = this.prepareHeader(jsonData[""]);
-            for (const [msgid, msgstrs] of Object.entries(jsonData)) {
-                if (msgid !== '') {
-                    // cockpit.js never looks at the first item
-                    msgstrs[0] = null;
+            const parsed = gettext_parser.po.parse(poData, 'utf8');
+            delete parsed.translations[""][""]; // second header copy
+
+            // cockpit.js only looks at "plural-forms" and "language"
+            const chunks = [
+                '{\n',
+                ' "": {\n',
+                `  "plural-forms": ${this.get_plural_expr(parsed.headers['plural-forms'])},\n`,
+                `  "language": "${parsed.headers.language}"\n`,
+                ' }'
+            ];
+            for (const [msgctxt, context] of Object.entries(parsed.translations)) {
+                const context_prefix = msgctxt ? msgctxt + '\u0004' : ''; /* for cockpit.ngettext */
+
+                for (const [msgid, translation] of Object.entries(context)) {
+                    if (translation.comments.flag === 'fuzzy')
```

Comment:
Is `===` correct? there could be more flags, doesn't that need to be a substring match?

Reply:
It didn't seem to change anything for our current set of `.po` files.  I'll do a follow-up.